plugins/inputs/ublox/ublox.go

```diff
@@ -79,15 +80,34 @@ func (s *UbloxDataCollector) Gather(acc telegraf.Accumulator) error {
 
 	if s.lastPos != nil {
 		metrics := make(map[string]interface{})
+		sensors := make(map[string]interface{})
+
 		metrics["active"] = s.lastPos.Active
 		metrics["lon"] = s.lastPos.Lon
 		metrics["lat"] = s.lastPos.Lat
+		metrics["horizontal_acc"] = s.lastPos.HorizontalAcc
+
 		metrics["heading"] = s.lastPos.Heading
+		metrics["heading_of_motion"] = s.lastPos.HeadingOfMotion
+		metrics["heading_acc"] = s.lastPos.HeadingAcc
+		metrics["heading_is_valid"] = s.lastPos.HeadingIsValid
+
+		metrics["speed"] = s.lastPos.Speed
+		metrics["speed_acc"] = s.lastPos.SpeedAcc
+
 		metrics["pdop"] = s.lastPos.Pdop
+		metrics["sat_num"] = s.lastPos.SatNum
+		metrics["fix_type"] = s.lastPos.FixType
 
 		if s.lastPos.FusionMode != None {
 			metrics["fusion_mode"] = s.lastPos.FusionMode
 		}
+		for i := 0; i*4 < len(s.lastPos.Sensors); i++ {
+			sensors[fmt.Sprintf("s%d_status1", i)] = s.lastPos.Sensors[i*4+0]
+			sensors[fmt.Sprintf("s%d_status2", i)] = s.lastPos.Sensors[i*4+1]
+			sensors[fmt.Sprintf("s%d_freq", i)] = s.lastPos.Sensors[i*4+2]
+			sensors[fmt.Sprintf("s%d_faults", i)] = s.lastPos.Sensors[i*4+3]
+		}
```

Comment:
As far as it'll be added to new metric it will not solve the huge amount of fields. Let's change the schema:
```suggestion
		for i := 0; i*4 < len(s.lastPos.Sensors); i++ {
			tags := map[string]interface{}{ "name": fmt.Sprintf("Sensor %d", i)}

			sensors["s_status1"] = s.lastPos.Sensors[i*4+0]
			sensors["s_status2"] = s.lastPos.Sensors[i*4+1]
			sensors["s_freq"] = s.lastPos.Sensors[i*4+2]
			sensors["s_faults"] = s.lastPos.Sensors[i*4+3]

			acc.AddFields("ublox-data-sensors", sensors, tags)
		}
```

It'd be worth to start writing tests for such changes
They're not hard to write - some test examples are there `plugins/inputs/smart/smart_test.go`

Reply:
could you make that changes?